src/inventorymanager.cpp

```diff
@@ -384,9 +389,13 @@ void IMoveAction::apply(InventoryManager *mgr, ServerActiveObject *player, IGame
 			src_can_take_count = dst_can_put_count = 0;
 	} else {
 		// Take from one inventory, put into another
+		int src_item_count = src_item.count;
+		if (caused_by_move_somewhere)
```

Comment:
If inside the move somewhere loop, temporarily set this to the move count.

Reply:
(In conflict with my comment above)
This should be documented/commented. Suggestion:
```
// When moving somewhere: temporarily use the actual movable stack
// size to ensure correct callback execution.
```